enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,871 @@
+
+### Managing settings for operators installed by cluster-version-operator
+
+The alternatives in this subsection cover the need to change the
+settings for the static manifests currently managed by the
+cluster-version-operator. These approaches would only apply to the
+operators themselves, and not their operands.
+
+1. Use a cluster profile to change the resource requests for control
+   plane operators
+
+   We could create a separate cluster profile with different resource
+   requests for our control plane components.
+
+   This would be quite a lot of work to implement and test.
+
+   We may eventually need different profiles for different customers,
+   further multiplying the amount of work.
+
+   As an organization, we are trying to keep the number of profiles
+   small as a general rule. The discussion about using a cluster
+   profile for deploying single production instances provides some
+   useful background. See
+   [#504](https://github.com/openshift/enhancements/pull/504),
+   [#560](https://github.com/openshift/enhancements/pull/560), and
+   [single-node-deployment-approach](single-node/production-deployment-approach.md).
+
+
+   Profile settings only apply to the cluster operators Deployments,
+   and changing the resource requests in the static manifests does not
+   signal to the operators that they need to change the settings for
+   their operands.
+
+2. Lower the resource requests for cluster operators statically in the
+   default configuration
+
+   We could apply changes to the default resource requests in the
+   static manifests used by all deployments, instead of adding one or
+   more new cluster profiles.
+
+   This would affect all clusters, and if we are too aggressive with
+   the changes we may lower values too much and cause important
+   cluster operators to be starved for resources, triggering degraded
+   cluster performance.
+
+   The static settings only apply to the cluster operators
+   Deployments, and changing them does not signal to the operators
+   that they need to change the settings for their operands.
+
+   With lowered CPU requests, the consumption of some operators may no
+   longer match the requests closely, especially for periodically
+   "bursty" workloads. One way to address that is by restructuring
+   some operators to use secondary Jobs to run some expensive
+   operations. For example, the insights operator could schedule a Job
+   to collect and push data. The operator would consume relatively few
+   resources, and the Job would be ephemeral. We will need to perform
+   significant analysis to identify opportunities for these sort of
+   changes.
+
+   Some [work to reduce default
+   requests](https://bugzilla.redhat.com/show_bug.cgi?id=1920159) has
+   been done in the 4.8 release.
+
+3. Have a new operator in the release payload perform the scaling for
+   control plane operators
+
+   It would use the new API field to determine how to scale (see above).
```

Comment:
We already have some operators for it - https://github.com/openshift/cluster-resource-override-admission-operator, we can improve it and add new API fields.

Reply:
OK, so we can modify this section to say it doesn't need to be in the release payload but that it would need to be installed separately and link to the existing operator.

We'll also need to say that each openshift operator will need to make the appropriate adjustments to ignore changes to the resource requests to avoid contention with the cluster-resource-override-admission web hook. It looks like the way to enable the admission web hook is to add a label to the namespace and then update the code to ignore resource requests on operands when comparing for equality. Is that right?

Reply:
@cynepco3hahue  The latest draft elevates the cluster-resource-override-admission-operator to the primary approach being proposed. Let me know what you think.